editor/src/components/editor/store/dispatch.tsx

```diff
@@ -105,168 +105,195 @@ function processAction(
   action: EditorAction,
   spyCollector: UiJsxCanvasContextData,
 ): EditorStoreUnpatched {
+  let shouldProcessAction: boolean = true
+  // By default when the current user does not own the project and the action is non-transient prevent actions from running...
+  if (editorStoreUnpatched.projectServerState.isMyProject === 'no' && !isTransientAction(action)) {
+    // ...Unless they're more critical to the running of the editor in this case.
+    const allowedNonOwnerAction =
+      action.action === 'UPDATE_FROM_WORKER' ||
+      action.action === 'UPDATE_TOP_LEVEL_ELEMENTS_FROM_COLLABORATION_UPDATE'
+    shouldProcessAction = allowedNonOwnerAction
+  }
+  // When the current user does own the project...
+  if (editorStoreUnpatched.projectServerState.isMyProject === 'yes') {
+    // ...Disallow these updates as they're coming from non-owners.
+    shouldProcessAction = action.action !== 'UPDATE_TOP_LEVEL_ELEMENTS_FROM_COLLABORATION_UPDATE'
+  }
+
```

Comment:
maybe nit, but I'd change this to
`const shouldProcessAction = isMyProject || isTransientAction || allowedNonOwnerAction`

and not even dispatch UPDATE_TOP_LEVEL_ELEMENTS_FROM_COLLABORATION_UPDATE if the update doesn't come from a project owner

Reply:
The latter bit is a bit nasty as it hangs off a callback that we assign during the `LOAD` action, at which point it would likely not be known if the user is the owner because that fires asynchronously.

Reply:
in that case, you could move it inside `UPDATE_TOP_LEVEL_ELEMENTS_FROM_COLLABORATION_UPDATE` to just return the current editor

Reply:
Fixed up in 889fd11d1.